packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -545,7 +554,7 @@ class Navigator extends StatefulWidget {
   /// The new route and the previous route (if any) are notified (see
   /// [Route.didPush] and [Route.didChangeNext]). If the [Navigator] has any
   /// [Navigator.observers], they will be notified as well (see
-  /// [NavigatorObserver.didPush]).
+  /// [Navigator.willPush] and [NavigatorObserver.didPush]).
```

Comment:
i assume you mean NavigatorObserver.willPush? (similar below a few times)

Reply:
Maybe "preloading" isn"t the best word.

We have routes that are built using data that is updated when the current Route changes. We want to make sure that whatever is holding this data is notified before the Route is actually built.

"didPush" might not be the right hook for that because it's unclear if the Route was already built when we get that notification, or even whether the answer is an implementation detail that we shouldn't rely on.